### PR TITLE
Update to latest spec

### PR DESCRIPTION
**Pull request by Sebmaster**
There's a problem with our `URLSearchParams` constructor. We explicitly check the objects prototype there, which is not correct, I think, which then fails one of the w3c tests.

### CODE REVIEW

**Comment by domenic** [comment]
First commit should probably include https://github.com/jsdom/whatwg-url/tree/host-parser-tests

Commit message format might in general be best to follow recent precedent with "Spec update:" and link to the pull request.

**Comment by domenic** [comment]
Oh and we should update the readme spec hash too

**Comment by Sebmaster** [comment]
Added the readme fixes. Sadly most of the host parser tests don't parse and that file is also missing from the w3c repo...

**Comment by TimothyGu** [comment]
@Sebmaster I created the following patch to make the DOMException somewhat cleaner and more IDL. It is rebased on faad8f1e9ae2a7af79030c9f1391d82eacf47d76 though.

[0001-More-accurate-DOMException-shim.patch](https://github.com/jsdom/whatwg-url/files/1181739/0001-More-accurate-DOMException-shim.txt)

**Comment by TimothyGu** [comment]
@domenic The host parser test exposes some bugs in tr46... I'll get to them today.

**Comment by Sebmaster** [comment]
> I created the following patch to make the DOMException somewhat cleaner and more IDL. It is rebased on faad8f1 though.

Do you think there's a point in going with this more complicated one? We just want to get the w3c tests passing, it doesn't have to be standards compliant.

> The host parser test exposes some bugs in tr46... I'll get to them today.

That's amazing 🙇

File: src/url-state-machine.js [hunk 1]
```diff
@@ -364,7 +364,12 @@ function parseHost(input, isSpecialArg) {
   }
 
   const domain = percentDecode(Buffer.from(input)).toString();
-  const asciiDomain = tr46.toASCII(domain, false, tr46.PROCESSING_OPTIONS.NONTRANSITIONAL, false);
+  const asciiDomain = tr46.toASCII(domain, {
+    checkBidi: true,
+    checkJoiners: true,
+    verifyDNSLength: true,
+    useSTD3ASCIIRules: true
+  });
```
Comment (Sebmaster) [inline src/url-state-machine.js]: Fixed.

**Comment by TimothyGu** [comment]
@Sebmaster I suppose not, but I don't feel like dealing with that code again if the WPT uses more functionality of DOMException (which I'm still not sure why it was added as a test case in the first place).

**Comment by Sebmaster** [comment]
> I suppose not, but I don't feel like dealing with that code again if the WPT uses more functionality of DOMException

I think @domenic's planning to move DOMException to its own package, so optimally we'd be fine with using this in the interim and then switch to the package when that's done.

**Comment by TimothyGu** [comment]
@Sebmaster Okay. I'd be more than okay with that.

**Comment by TimothyGu** [comment]
https://github.com/Sebmaster/tr46.js/pull/12

**Comment by Sebmaster** [comment]
Okay, tests are sadly broken due to an unrelated issue, but should be ready for re-review and merge @TimothyGu.

When executing `npm test`, make sure `npm ls tr46` gives you
```
+-- jsdom@11.2.0
| `-- whatwg-url@6.1.0
|   `-- tr46@0.0.3
`-- tr46@1.0.1
```

otherwise it won't work.

Problem is that the new tr46 included in root overwrites jsdom's version and jsdom's whatwg-url access a removed property.

**Comment by TimothyGu** [comment]
Sounds like an npm@5 bug…

**Comment by Sebmaster** [comment]
> Sounds like an npm@5 bug…

Yeah that's what I'm wondering. I think a parent overwriting a childs dependencies might be intended? I'm not quite sure though and it sucks I found nothing to disable it.

**Comment by Sebmaster** [comment]
Switching to yarn fixed it. So that's a thing now.

**Pull request merged**